gammapy/maps/wcs/geom.py

```diff
@@ -811,7 +816,7 @@ def solid_angle(self):
 
     @lazyproperty
     def _solid_angle(self):
-        coord = self.get_coord(mode="edges").skycoord
+        coord = self.to_image().get_coord(mode="edges").skycoord
```

Comment:
Now it is applied only `if self.is_regular`
I'm strongly against keeping the old behaviour as default for regular maps because there is no gain from that, just a waste of memory.